File: ui/theme.py
"""Menu-specific color and size constants."""

# -- background -------------------------------------------------------------
MENU_BG = (12, 12, 20)

# -- title -------------------------------------------------------------------
TITLE_COLOR = (220, 220, 240)
TITLE_SHADOW_COLOR = (40, 40, 60)
SUBTITLE_COLOR = (140, 140, 170)
TITLE_FONT_SIZE = 64
SUBTITLE_FONT_SIZE = 22

# -- buttons -----------------------------------------------------------------
BTN_NORMAL = (35, 35, 50)
BTN_HOVER = (55, 55, 75)
BTN_PRESS = (25, 25, 40)
BTN_TEXT = (220, 220, 240)
BTN_BORDER = (80, 80, 110)
BTN_WIDTH = 260
BTN_HEIGHT = 44
BTN_FONT_SIZE = 20
BTN_BORDER_RADIUS = 6

# -- back button -------------------------------------------------------------
BACK_BTN_SIZE = 36
BACK_BTN_MARGIN = 12
BACK_BTN_COLOR = (180, 180, 200)

# -- dropdown ----------------------------------------------------------------
DD_BG = (30, 30, 45)
DD_HOVER = (50, 50, 70)
DD_BORDER = (70, 70, 100)
DD_TEXT = (200, 200, 220)
DD_WIDTH = 220
DD_HEIGHT = 32
DD_FONT_SIZE = 16

# -- text input ---------------------------------------------------------------
TI_BG = (30, 30, 45)
TI_ACTIVE_BG = (40, 40, 55)
TI_BORDER = (70, 70, 100)
TI_ACTIVE_BORDER = (80, 140, 255)
TI_TEXT = (200, 200, 220)
TI_PLACEHOLDER = (100, 100, 120)

# -- slider ------------------------------------------------------------------
SL_TRACK_COLOR = (50, 50, 70)
SL_FILL_COLOR = (80, 140, 255)
SL_HANDLE_COLOR = (200, 200, 220)
SL_TEXT_COLOR = (180, 180, 200)
SL_WIDTH = 220
SL_HEIGHT = 8
SL_HANDLE_RADIUS = 8
SL_FONT_SIZE = 16

# -- checkbox ----------------------------------------------------------------
CB_BOX = (35, 35, 50)
CB_CHECK = (80, 255, 120)
CB_BORDER = (80, 80, 110)
CB_DISABLED = (60, 60, 70)

# -- toggle group ------------------------------------------------------------
TG_ACTIVE = (80, 140, 255)
TG_INACTIVE = (40, 40, 55)
TG_BORDER = (70, 70, 100)
TG_TEXT = (220, 220, 240)
TG_FONT_SIZE = 16

# -- results screen ----------------------------------------------------------
RESULT_VICTORY_COLOR = (80, 255, 120)
RESULT_DEFEAT_COLOR = (255, 80, 80)
RESULT_DRAW_COLOR = (200, 200, 200)
RESULT_FONT_SIZE = 72
RESULT_SUB_FONT_SIZE = 24

# -- guides / unit overview --------------------------------------------------
SIDEBAR_BG = (20, 20, 32)
SIDEBAR_WIDTH = 200
SIDEBAR_BTN_HEIGHT = 36
CONTENT_BG = (16, 16, 26)
CONTENT_TEXT = (200, 200, 220)
CONTENT_HEADING = (220, 220, 240)
CONTENT_FONT_SIZE = 16
HEADING_FONT_SIZE = 24

# -- background animation (main menu) ---------------------------------------
BG_DOT_RADIUS = 4
BG_DOT_SPEED = 20
BG_DOT_COUNT = 30

# -- line graph / stats screen -----------------------------------------------
GRAPH_BG = (20, 20, 32)
GRAPH_GRID = (40, 40, 55)
GRAPH_AXIS_TEXT = (140, 140, 160)
GRAPH_LINE_T1 = (80, 160, 255)
GRAPH_LINE_T2 = (255, 90, 90)
GRAPH_FILL_T1 = (80, 160, 255, 30)
GRAPH_FILL_T2 = (255, 90, 90, 30)
GRAPH_TITLE_COLOR = (200, 200, 220)
GRAPH_FONT_SIZE = 14
SCORE_FONT_SIZE = 28
SCORE_T1_COLOR = (100, 180, 255)
SCORE_T2_COLOR = (255, 120, 120)
STATS_HEADER_FONT_SIZE = 40

# -- debug performance graph ------------------------------------------------
DEBUG_LINE_COLORS = [
    (255, 215, 0),     # step_ms — gold/yellow
    (200, 200, 200),   # commands — light gray
    (80, 220, 120),    # entity_update — green
    (180, 100, 255),   # ai_step — purple
    (255, 165, 60),    # capture — orange
    (80, 160, 255),    # targeting_build — blue
    (100, 140, 220),   # tgt_qf_sync — steel blue
    (130, 190, 255),   # tgt_populate — light cornflower
    (80, 220, 220),    # combat — cyan
    (255, 130, 180),   # spawn — pink
    (255, 255, 140),   # cleanup — pale yellow
    (160, 255, 80),    # physics — lime
    (120, 180, 60),    # phys_array_build — olive
    (200, 255, 100),   # phys_unit_collisions — bright lime
    (100, 200, 140),   # phys_obstacle_push — teal-green
    (180, 220, 60),    # phys_writeback — yellow-green
    (140, 240, 180),   # phys_clamp — mint
]
STATS_SUB_FONT_SIZE = 18
BUILD_ORDER_RADIUS = 6

# -- general layout ----------------------------------------------------------
MENU_WIDTH = 800
MENU_HEIGHT = 600

File: web/src/ui/theme.ts
// Menu/widget colors and sizes — ported from ui/theme.py so the canvas-drawn
// UI matches the pygame client. RGB tuples become CSS rgb() strings via rgb().

export type RGB = [number, number, number];
export type RGBA = [number, number, number, number];

export function rgb(c: RGB | RGBA): string {
  if (c.length === 4) return `rgba(${c[0]},${c[1]},${c[2]},${c[3] / 255})`;
  return `rgb(${c[0]},${c[1]},${c[2]})`;
}

// -- background --
export const MENU_BG: RGB = [12, 12, 20];

// -- title --
export const TITLE_COLOR: RGB = [220, 220, 240];
export const TITLE_SHADOW_COLOR: RGB = [40, 40, 60];
export const SUBTITLE_COLOR: RGB = [140, 140, 170];
export const TITLE_FONT_SIZE = 64;
export const SUBTITLE_FONT_SIZE = 22;

// -- buttons --
export const BTN_NORMAL: RGB = [35, 35, 50];
export const BTN_HOVER: RGB = [55, 55, 75];
export const BTN_PRESS: RGB = [25, 25, 40];
export const BTN_TEXT: RGB = [220, 220, 240];
export const BTN_BORDER: RGB = [80, 80, 110];
export const BTN_DISABLED_TEXT: RGB = [110, 110, 130];
export const BTN_WIDTH = 260;
export const BTN_HEIGHT = 44;
export const BTN_FONT_SIZE = 20;
export const BTN_BORDER_RADIUS = 6;

// -- back button --
export const BACK_BTN_SIZE = 36;
export const BACK_BTN_MARGIN = 12;
export const BACK_BTN_COLOR: RGB = [180, 180, 200];

// -- dropdown --
export const DD_BG: RGB = [30, 30, 45];
export const DD_HOVER: RGB = [50, 50, 70];
export const DD_BORDER: RGB = [70, 70, 100];
export const DD_TEXT: RGB = [200, 200, 220];
export const DD_WIDTH = 220;
export const DD_HEIGHT = 32;
export const DD_FONT_SIZE = 16;

// -- text input --
export const TI_BG: RGB = [30, 30, 45];
export const TI_ACTIVE_BG: RGB = [40, 40, 55];
export const TI_BORDER: RGB = [70, 70, 100];
export const TI_ACTIVE_BORDER: RGB = [80, 140, 255];
export const TI_TEXT: RGB = [200, 200, 220];
export const TI_PLACEHOLDER: RGB = [100, 100, 120];

// -- slider --
export const SL_TRACK_COLOR: RGB = [50, 50, 70];
export const SL_FILL_COLOR: RGB = [80, 140, 255];
export const SL_HANDLE_COLOR: RGB = [200, 200, 220];
export const SL_TEXT_COLOR: RGB = [180, 180, 200];
export const SL_WIDTH = 220;
export const SL_HEIGHT = 8;
export const SL_HANDLE_RADIUS = 8;
export const SL_FONT_SIZE = 16;

// -- checkbox --
export const CB_BOX: RGB = [35, 35, 50];
export const CB_CHECK: RGB = [80, 255, 120];
export const CB_BORDER: RGB = [80, 80, 110];
export const CB_DISABLED: RGB = [60, 60, 70];

// -- toggle group --
export const TG_ACTIVE: RGB = [80, 140, 255];
export const TG_INACTIVE: RGB = [40, 40, 55];
export const TG_BORDER: RGB = [70, 70, 100];
export const TG_TEXT: RGB = [220, 220, 240];
export const TG_FONT_SIZE = 16;

// -- sidebar (guides / unit overview) --
export const SIDEBAR_BG: RGB = [20, 20, 32];
export const SIDEBAR_WIDTH = 200;
export const SIDEBAR_BTN_HEIGHT = 36;

// -- content / headings --
export const CONTENT_TEXT: RGB = [200, 200, 220];
export const CONTENT_HEADING: RGB = [220, 220, 240];
export const CONTENT_FONT_SIZE = 16;
export const HEADING_FONT_SIZE = 24;

// -- create-lobby panels (from screens/create_lobby.py) --
export const PANEL_BG: RGB = [18, 18, 28];
export const PANEL_BORDER: RGB = [42, 42, 62];
export const HDR_COLOR: RGB = [160, 160, 185];
export const ONLINE_COLOR: RGB = [100, 255, 140];

// -- player colors (matches config PLAYER_COLORS) --
export const PLAYER_COLORS: RGB[] = [
  [80, 140, 255],
  [80, 220, 160],
  [255, 80, 80],
  [255, 160, 60],
  [180, 80, 220],
  [80, 220, 220],
  [220, 220, 80],
  [220, 80, 160],
];

// -- background animation (main menu) --
export const BG_DOT_RADIUS = 4;
export const BG_DOT_SPEED = 20;
export const BG_DOT_COUNT = 30;

// Font family used for all canvas text. pygame uses SysFont(None, size); a
// system sans-serif stack is the closest faithful match in the browser.
export const FONT_FAMILY =
  'system-ui, -apple-system, "Segoe UI", Roboto, Helvetica, Arial, sans-serif';
